src/api/services/enrichment/enrichment.js

```diff
@@ -0,0 +1,86 @@
+import * as fs from 'fs';
+import path from 'path';
+
+export const createEnrichmentRule = async ctx => {
+    const enrichment = ctx.request.body;
+    if (enrichment.hasOwnProperty('rule')) {
+        return enrichment;
+    }
+
+    if (!enrichment.webServiceUrl || !enrichment.sourceColumn) {
+        throw new Error(`Missing parameters`);
+    }
+
+    try {
+        const excerpt = await ctx.dataset.getExcerpt();
+        const sourceData = excerpt[0][enrichment.sourceColumn];
+        let data;
+        try {
+            data = JSON.parse(sourceData);
+        } catch {
+            data = sourceData;
+        }
+
+        let rule = getEnrichmentRuleModel(data, enrichment);
+
+        return {
+            name: enrichment.name,
+            advancedMode: enrichment.advancedMode,
+            rule: rule,
+        };
+    } catch (e) {
+        console.log('Error:', e.stack);
```

Comment:
I think there is something wrong here, if we catch an erro, then we return nothing, maybe we should just let the error propagate to the top level?